webpack.config.js

```diff
@@ -17,7 +17,13 @@ if (DAPTrackingURL) {
   connectSrcURLs.push(DAPTrackingURL);
 }
 if (process.env.DATA_UPLOAD_BUCKET) {
-  connectSrcURLs.push(`https://${process.env.DATA_UPLOAD_BUCKET}.s3.amazonaws.com`);
+  if (Array.isArray(process.env.DATA_UPLOAD_BUCKET)) {
+    // if DATA_UPLOAD_BUCKET is an array
+    connectSrcURLs.push(...process.env.DATA_UPLOAD_BUCKET.map(dataBucket => (`https://${dataBucket}.s3.amazonaws.com`)));
```

Comment:
is this going to work? isn't `process.env.XXX` will always be strings?

how are you going to mount this env vars into portal pod? is there an accompanying `cloud-automation` PR?

Reply:
from what i see DATA_UPLOAD_BUCKET is configured in fence-config-public.yaml i was just going to set it to be an array like 
`DATA_UPLOAD_BUCKET: ['qa-covid19-databucket-gen3', covd-map-occ-prc-qa]`

the new code will then add these new values as well as not breaking the old functionality. What do you think?

Reply:
First, according to https://github.com/uc-cdis/fence/blob/e31d29d5533683d91996f909f4b364e0cf798168/fence/config-default.yaml#L591-L594 `DATA_UPLOAD_BUCKET` is just the name of one bucket, I think Fence might not like it if you set it as an array

Second, like I said `process.env.XXX` will always be strings. So even if you change `DATA_UPLOAD_BUCKET` to be an array in fence config, when passed into the Portal pod it will becomes `"['qa-covid19-databucket-gen3', 'covd-map-occ-prc-qa']"`. And you have to do something manually with `. replace()` and `.split()`

You need to test this in k8s to see if your portal pods is mounting something as you have expected, and is parsing it correctly

Reply:
And fundamentally why would you put `covd-map-occ-prc-qa` as one of Fence's `DATA_UPLOAD_BUCKET`? It doesn't sound right since we are not going to upload anything to there right? Just pulling things down?

IMO an more appropriate would be to define another optional env var like `CONNECTED_S3_BUCKET`, put it some where in the manifest (like in the `portal` section), and mount it into portal pods as env var

Example https://github.com/uc-cdis/cloud-automation/blob/master/kube/services/portal/portal-deploy.yaml#L98-L103

Reply:
thanks for the suggestion will implement it that way, branching this off as its own separate jira COV-996

and will remove the above code from this PR